pkg/identity/cache.go

```diff
@@ -119,24 +119,39 @@ func LookupIdentity(lbls labels.Labels) *Identity {
 // LookupReservedIdentityByLabels looks up a reserved identity by its labels and
 // returns it if found. Returns nil if not found.
 func LookupReservedIdentityByLabels(lbls labels.Labels) *Identity {
-	// If there is only one label with the "reserved" source and a well-known
-	// key, return the well-known identity for that key.
-	if len(lbls) != 1 {
-		return nil
-	}
+	var (
+		idty                *Identity
+		containsNonReserved bool
+	)
 	for _, lbl := range lbls {
-		if lbl.Source != labels.LabelSourceReserved {
-			return nil
-		}
-		if id := GetReservedID(lbl.Key); id != IdentityUnknown {
-			return LookupReservedIdentity(id)
+		switch {
+		// If the set of labels contain a fixed identity then and exists in
+		// the map of reserved IDs then return the identity of that reserved ID.
+		case lbl.Key == labels.LabelKeyFixedIdentity:
+			id := GetReservedID(lbl.Value)
+			if id != IdentityUnknown && IsUserReservedIdentity(id) {
+				return LookupReservedIdentity(id)
+			}
```

Comment:
In what case do we want to continue looking here? If there is a label with the key `labels.LabelKeyFixedIdentity` then we either find it and it is a user reserved identity or we return nil as we clearly can't derive the identity from it.

By continuing, we would potentially fall to another reserved label which is not what we want.

Reply:
Correct. Fixed.